scheduler/src/cook/tools.clj

```diff
@@ -249,6 +249,9 @@
                     (let [resource (keyword (name (:resource/type r)))]
                       (condp contains? resource
                         #{:cpus :mem :gpus} (assoc m resource (:resource/amount r))
+                        #{:disk} (assoc m :disk (cond-> {:request (:resource.disk/request r)}
+                                                        (:resource.disk/limit r) (assoc :limit (:resource.disk/limit r))
```

Comment:
Also, to pose a question, if a node does not have disk resources, but a job does declare a need for disk resources (or gets resources inserted by default), will putting this code here cause fenzo to fail to binpack on that node. (But, if tets pass, then obviously we're OK, but please make sure to test this on k8s and mesos and GKE.)  I think we're safe, but we don't want to break production.

Reply:
This is handled in our API validation. If a job declares a need for disk resources, we check that the pool (and thus compute cluster) has nodes with disk resources available. On k8s on-prem, the request/limit doesn't actually get enforced, but the jobs will pass. If a job requests disk resources on Mesos the job validation will fail.